src/java/voldemort/store/rocksdb/RocksDbStorageConfiguration.java

```diff
@@ -0,0 +1,115 @@
+package voldemort.store.rocksdb;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+import org.rocksdb.Options;
+import org.rocksdb.RocksDB;
+import org.rocksdb.util.SizeUnit;
+
+import voldemort.routing.RoutingStrategy;
+import voldemort.server.VoldemortConfig;
+import voldemort.store.StorageConfiguration;
+import voldemort.store.StorageEngine;
+import voldemort.store.StorageInitializationException;
+import voldemort.store.StoreBinaryFormat;
+import voldemort.store.StoreDefinition;
+import voldemort.utils.ByteArray;
+
+public class RocksDbStorageConfiguration implements StorageConfiguration {
+
+    static {
+        RocksDB.loadLibrary();
+    }
+
+    private final int lockStripes;
+
+    public static final String TYPE_NAME = "rocksdb";
+
+    private static Logger logger = Logger.getLogger(RocksDbStorageConfiguration.class);
+
+    private final VoldemortConfig voldemortconfig;
+
+    private Map<String, RocksDbStorageEngine> stores = new HashMap<String, RocksDbStorageEngine>();
+
+    public RocksDbStorageConfiguration(VoldemortConfig config) {
+        /**
+         * - TODO 1. number of default locks need to debated. This default is
```

Comment:
TODO still relevant? in general we should set the striped lock count to a multiple of number NIO server threads (the idea is there will be only as much locking as number of threads)